Sources/MapboxCoreNavigation/EventDetails.swift

```diff
@@ -52,9 +52,22 @@ struct PerformanceEventDetails: EventDetails {
     }
 }
 
-struct NavigationEventDetails: EventDetails {
-    let audioType: String = AVAudioSession.sharedInstance().audioType
-    let applicationState: UIApplication.State = {
+protocol NavigationEventDetails: EventDetails {
```

Comment:
A protocol may be the best solution, but in case it wasn’t considered, another approach would be composition: keep this struct intact but add a property that could be either active-specific or passive-specific.

Reply:
I think the protocol is fine here, I didn't find another approach that seems okish for me.